lib/menubar/utils.lua

```diff
@@ -279,20 +267,21 @@ function utils.lookup_icon_uncached(icon_file)
         return false
     end
 
-    if icon_file:sub(1, 1) == '/' and is_format_supported(icon_file) then
+    local icon_file_ext = icon_file:gsub(".*%.","")
```

Comment:
Uhm. This does not behave in the same way as the old code. Previously a file called "png" was rejected:
```
$ lua -e 'print(("png"):match("%.png"))'
nil
```
now it is accepted:
```
$ lua -e 's = {png=1} print(s[("png"):gsub(".*%.","")])'
1
```

Reply:
The old way used the function I removed...
```
icon_formats = { "png", "xpm", "svg" }

icon_file = "file.png"

local function is_format_supported(icon_file)
    for _, f in ipairs(icon_formats) do            -- May try all of these { "png", "xpm", "svg" }
        if icon_file:match('%.' .. f) then         -- "file.png":match(".png")
            return true
        end
    end
    return false
end

if is_format_supported("file.png") then print("Match!") end

> Match!
```

The new way simply uses the extension as an index (is true simply if it exists)...
```
supported_icon_formats = { ["png"] = 1, ["xpm"] = -1, ["svg"] = 'a' }

icon_file = "file.png"

icon_file_ext = icon_file:gsub(".*%.","")                            -- icon_file_ext = "file.png":gsub("file.","") = "png"
if supported_icon_formats[icon_file_ext] then print("Match!") end    -- supported_icon_formats["png"] = 1

> Match!
```

Tested here... https://www.lua.org/cgi-bin/demo

Reply:
What I am trying to say: If you change `file.png` to `png`, then the old code will not print anything, but the new one will still print `Match!`. So, there is a change in behaviour here.

Reply:
Just saw your follow up... I think it is safe to say that the use-case is for a filename ("name,ext") and not just an extension - The comments note that a filename without an extension would be acceptable but I think it's safe to say that it isn't expecting someone to just to an extension

Nice boundary value test however